source/tutorials/learning-journey/sharing-dependencies.md

```diff
@@ -0,0 +1,198 @@
+You could certainly copy the build dependencies from `default.nix` and copy them into `shell.nix`, but this is less than ideal.
+Your build dependencies are defined in multiple places, and aside from repeating yourself there's now the possiblity that the dependencies in `default.nix` and `shell.nix` may fall out of sync.
+
+There is a better way!
+
+## Getting started
+
+Create a directory called `shared_project` and enter it:
+```console
+$ mkdir shared_project
+$ cd shared_project
+```
+
+You'll be creating a Python web application as an example project, but don't worry, you'll be given all of the code you need and won't need to know Python to proceed.
+
+Create a file `app.py` with the following contents:
+```python
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello_world():
+    return "<p>Hello, World!</p>"
+```
+This creates a web application that returns `<p>Hello, World!</p>` on the `/` route.
+
+Next create a `pyproject.toml` file with the following contents:
+```toml
+[build-system]
+requires = ["setuptools", "setuptools-scm"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "shared_project"
+version = "0.0.1"
+```
+This file tells Python how to build the project.
+
+For the Nix part of the project you'll create two files: `build.nix` and `default.nix`.
+The actual build recipe will be in `build.nix` and `default.nix` will import this file to perform the build.
+
+First create a `build.nix` file like this:
+```nix
+{
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication {
+  pname = "shared_project";
+  version = "0.0.1";
+  format = "pyproject";
+  src = builtins.path { path = ./.; name = "shared_project_source"; };
+  propagatedBuildInputs = with python3Packages; [
+    setuptools-scm
+    flask
+  ];
+}
+```
+The Nix expression in this file is a function that takes `python3Packages` as input and produces a derivation.
+This method of defining builds is a common design pattern in the Nix community, and is the format used throughout the `nixpkgs` repository.
+This particular derivation builds your Python application and ensures that `flask`, the library used to create the web application, is available at runtime for the application.
+
+Finally, create a `default.nix` that looks like this:
+```nix
+let
+  pkgs = import <nixpkgs> {};
+in
+  pkgs.callPackage ./build.nix {}
+```
+The `callPackage` function reads the expression in `build.nix` to determine which inputs it needs (in this case, python3Packages), then calls the expression with the inputs that were requested.
+You can read more about `callPackage` in the [Nix Pills][nix_pills_callpackage].
+
+[nix_pills_callpackage]: https://nixos.org/guides/nix-pills/callpackage-design-pattern.html
+
+If you should now be able to build the project with the `nix-build` command.
+If you now execute the `nix-shell` command you should also be put into a shell with the `python3` and `flask` commands available.
```

Comment:
But then `nix-shell -A build` would have to be used, or not mention it yet at all